### PR TITLE
Remove unnecessary TableRegistry references from JsonApiListener

### PR DESCRIPTION
**Pull request by Marlinc**

### CODE REVIEW

**Comment by ADmad** [comment]
Tests are failing

**Comment by jippi** [comment]
I **think** the behavior is strictly not the same as before

`is_a` "`checks if the object is of this class or has this class as one of its parents´" where the new strict equal will not look at the class hierarchy, which mean if someone extends the build-in association types, those will not be detected correctly.  

Using `instanceof ` or an interface would be more robust

**Comment by ceeram** [comment]
Does crud jsonapi listener need to support custom assocation classes? I think the decoupling from ORM is a good thing here personally. I am biased here though as we will use the listener with a non ORM datasource

**Comment by lorenzo** [comment]
The changes look good to me, but tests should pass

**Comment by bravo-kernel** [comment]
Making the listener produce JSON API for all sorts of datasources fits exactly into the megalomanic vision I had when I started work on this, just didn't have a clue where to start so all 👍 ❤️

**Comment by Marlinc** [comment]
I'm working on getting the tests to pass.

File: src/View/JsonApiView.php [hunk 1]
```diff
@@ -213,8 +213,27 @@ protected function _entitiesToNeoMerxSchema(array $entities)
                 throw new CrudException('JsonApiListener cannot not find Entity class ' . $entityName);
             }
 
+            $entityShortName = $entityName;
+            if (strpos($entityName, '\\') !== false) {
```
Comment (Marlinc) [inline src/View/JsonApiView.php]: The entity names that come from the `entiyClass` method are actual full class names. This block of code is necessary to turn those full class names back into the plugin split format which is required by `App::className`.

**Comment by Marlinc** [comment]
The tests are passing now :smile:

**Comment by lorenzo** [comment]
Great job!

**Comment by bravo-kernel** [comment]
@Marlinc I just checked the results in my test API and am seeing an undesired side-effect. The `type` node is now producing full path where it would previously display just `countries`. 

```json
{
  "data": [
    {
      "type": "app\\model\\entity\\countries",
      "id": "1",
      "attributes": {
        "code": "NL",
        "name": "The Netherlands"
    }
  ]
}
```

**Pull request merged**